reactioncommerce/reaction-component-library
# (refactor): 257 nnnnat dynamic locales

Resolves #257 
Impact: **minor**  
Type: **feature|refactor**

## Component
* Refactored the `AddressForm` component:
      *  To accept a `props.locales` object.
      *  To populate the Country and Region fields.
      *  Removed unnecessary props.
      *  Docs & Test clean up.
 * Refactored the `ShippingAddressCheckoutAction` component:
       *  To not provide or handle AddressForm locales anymore.
       *  Docs clean up.

## Styleguide
*  Added a few example locales JSON files
*  Created a `withLocales` HOC that can import locales data and provide it as props
*  Added `withLocales` to the `AddressForm` in the components-context so all components that use it will have locales available by default.

## Breaking changes
N/A

## Testing
1. Verify the `AddressForm` examples are working as expected.
2. Verify the `ShippingAddressCheckoutAction` examples are still working.
3. Verify the `CheckoutAction` example is still working.
